### PR TITLE
Promote VolumeAttachment to GA

### PR DESCRIPTION
**Pull request by jsafrane**
**What this PR does / why we need it**:
Copy VolumeAttachment from storage.k8s.io/v1beta1 to v1.

**Special notes for your reviewer**:
This is necessary for CSI to go to GA.

**Release note**:

```release-note
VolumeAttachment API is now GA
```

/sig storage
@kubernetes/sig-architecture-api-reviews

### CODE REVIEW

File: staging/src/k8s.io/api/storage/v1/types.go [hunk 1]
```diff
@@ -102,3 +102,110 @@ const (
...
+// VolumeAttachmentSource represents a volume that should be attached.
+// Right now only PersistenVolumes can be attached via external attacher,
+// in future we may allow also inline volumes in pods.
+// Exactly one member can be set.
+type VolumeAttachmentSource struct {
+	// Name of the persistent volume to attach.
+	// +optional
+	PersistentVolumeName *string `json:"persistentVolumeName,omitempty" protobuf:"bytes,1,opt,name=persistentVolumeName"`
+
+	// Placeholder for *VolumeSource to accommodate inline volumes in pods.
```
Comment (liggitt) [inline staging/src/k8s.io/api/storage/v1/types.go]: shouldn't the inline bits be added/finalized before promotion to v1?

Comment (jsafrane) [reply staging/src/k8s.io/api/storage/v1/types.go]: Would it help if I removed the comment? We still want plan inline volumes, but as alpha in 1.13 / 1.14. It's orthogonal to the rest of CSI features.

**Comment by liggitt** [comment]
I'm surprised CSI is being promoted to GA before we have the inline CSI stuff completed.

**Comment by jsafrane** [comment]
This is part of CSI GA in 1.13, https://github.com/kubernetes/features/issues/178

CSI has many features and inline volumes are tracked separately in https://github.com/kubernetes/features/issues/596.

@saad-ali, FYI

**Comment by spiffxp** [comment]
/milestone v1.13
I'm adding this to the v1.13 milestone since it relates to https://github.com/kubernetes/features/issues/178 which is currently planned for v1.13

**Pull request closed without merging**